### PR TITLE
Run tests on Chrome for Android on Linux systems using emulator

### PR DESCRIPTION
**Pull request by mithro**
(Also run on travis as gating browser.)

### CODE REVIEW

File: .travis-run.sh [hunk 1]
```diff
@@ -21,5 +21,9 @@ else
     exit 0
   fi
 
-  echo ./run-tests.sh $ARGS | bash || exit 1
+  if [ x$BROWSER == "xAndroid-Chrome" ]; then
```
Comment (mikelawther) [inline .travis-run.sh:24]: I know this isn't part of this review, but I needed to expand the whole file to try and understand the diff. Given that line 17 exits unconditionally, it seems unneeded to have an else on line 18.

File: .travis-run.sh [hunk 2]
```diff
@@ -21,5 +21,9 @@ else
...
+    echo ./run-tests-android.sh $ARGS | bash || exit 1
```
Comment (mikelawther) [inline .travis-run.sh:25]: I haven't seen this echo script.sh $ARGS | bash construct before. Does bash -c "./run-tests-android.sh $ARGS" work here?

File: .gitmodules [hunk 3]
```diff
@@ -12,4 +12,4 @@
 	url = https://github.com/ariya/esprima
 [submodule "test/browserdetect"]
 	path = test/browserdetect
-	url = https://github.com/ded/bowser.git
+	url = https://github.com/web-animations/bowser.git
```
Comment (mikelawther) [inline .gitmodules:15]: I don't see any bowser related changes in this CL?

Comment (mithro) [inline .gitmodules:15]: They are in the bowser repository and already pushed there. Currently have a pull request upstream for the changes. (https://github.com/ded/bowser/pull/34)

File: .travis-run.sh (hunk 2 above)
Comment (mithro) [inline .travis-run.sh:25]: bash -c format breaks the quoting :/

File: test/test-runner.html [hunk 4]
```diff
@@ -570,7 +574,7 @@ <h1 id=status>Pending</h1>
   // Deal with tests which are taking too long...
   for (var i in testStates['RUNNING']) {
     var test = testStates['RUNNING'][i];
-    if (now() - test.start > 30e3) {
+    if (now() - test.start > 300e3) {
```
Comment (mikelawther) [inline test/test-runner.html:577]: This seems like a long time. Is it because we're emulating Chrome on Android?

Comment (mithro) [inline test/test-runner.html:577]: Two primary reasons;
- We are using an emulator for Android
- Android Chrome is _much_ slower then desktop chrome on the polyfill (even on real devices).

The longest test goes from less then 4 seconds to taking around 90 seconds...

File: .travis-run.sh (hunk 2 above)
Comment (mikelawther) [inline .travis-run.sh:25]: Boo :(

Comment (mithro) [inline .travis-run.sh:25]: Yeah, bash quoting/unquoting and environment variables is super hard :/

File: .gitmodules (hunk 3 above)
Comment (mikelawther) [inline .gitmodules:15]: OK.

File: test/test-runner.html (hunk 4 above)
Comment (mikelawther) [inline test/test-runner.html:577]: Not for this patch, but it'd be nice if we didn't make other faster systems timeout at the speed of the slowest system.

Comment (mithro) [inline test/test-runner.html:577]: Even better if we made the slowest system faster :)

**Comment by mikelawther** [comment]
lgtm.

**Pull request merged**